src/Symfony/Component/Serializer/Normalizer/NormalizerWithCacheableSupportsMethod.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+/**
+ * Lets the normalizer advertise if the returns of its "supportsNormalization" method is cacheable.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+interface NormalizerWithCacheableSupportsMethod
```

Comment:
What would you think about renaming `NormalizerWithCacheableSupportsMethod ` to `CacheableNormalizer`?

Reply:
It's the `supports` method that is cacheable not the normalizer itself if I'm not mistaking.

Reply:
`CacheableNormalizer` is misleading (the result of the normalization isn't cacheable itself, it's only the call to the `supports` method).

But the performance improvement is so huge that I think that caching should be the default behavior, and disabling the cache should be opt-out. Having a normalizer with a `supports*` method result that isn't cacheable is a edge case.

I would like to make the cache opt-out instead of opt-in, but it will be a BC break... On the other hand it improves DX and performance for everybody.

Reply:
Missing `Interface` suffix.
What about `CacheableSupportsInterface`
I'd also add the supports method in the interface if possible.
Also, what about using constants for possible values?
VARY_BY_TYPE, etc.?